src/libos/src/process/do_vfork.rs

```diff
@@ -84,21 +112,53 @@ pub fn do_vfork(mut context: *mut CpuContext) -> Result<isize> {
         mem::replace(&mut *current_file_table, new_file_table)
     };
     if let Some(_) = vfork_file_tables.insert(parent_pid, parent_file_table) {
```

Comment:
I think it makes more sense to use `try_insert`.

Reply:
Rewrite the logic and put file table saving after stop all other threads.